src/renderer/components/+namespaces/add-namespace-dialog.tsx

```diff
@@ -33,20 +33,18 @@ export class AddNamespaceDialog extends React.Component<Props> {
     this.namespace = "";
   };
 
-  close = () => {
-    AddNamespaceDialog.close();
-  };
-
   addNamespace = async () => {
     const { namespace } = this;
     const { onSuccess, onError } = this.props;
 
     try {
-      await namespaceStore.create({ name: namespace }).then(onSuccess);
-      this.close();
+      const created = await namespaceStore.create({ name: namespace });
+
+      onSuccess?.(created);
+      AddNamespaceDialog.close();
```

Comment:
should this be in a `finally` block instead?

Reply:
No, because the user might want to fix what caused the error to happen and try again.